### PR TITLE
feat(vault-jwt): Add Vault JWT/OIDC module

### PR DESCRIPTION
**Pull request by matifali**
This module makes use of existing OIDC access token to authenticate with Vault. It requires setting up a Vault [JWT/OIDC](https://developer.hashicorp.com/vault/docs/auth/jwt) auth with the same OIDC provider used with Coder.

This module should let users get non interactive authentication with vault. 

I tested this with Okta.

### CODE REVIEW

File: vault-jwt/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+vault kv get -namespace=coder -mount=secrets coder
+```
+
+or using the Vault API:
+
+```shell
+curl -H "X-Vault-Token: ${VAULT_TOKEN}" -X GET "${VAULT_ADDR}/v1/coder/secrets/data/coder"
+```
+
+![Vault login](#)
```
Comment (Parkreiner) [inline vault-jwt/README.md:37]: Was this supposed to be an image URL?

**Review by Parkreiner — COMMENTED** [review]
I looked through the files, and I think they look good. The only thing I'm not sure about is the `run.sh` file, because I'm definitely not a Bash expert

Don't feel comfortable approving yet, since the Bash script seems to be where the most logic lives. Could we tag in someone like @mafredri to look things over?

File: vault-jwt/README.md (hunk 1 above)
Comment (matifali) [reply vault-jwt/README.md:37]: Yes. I will take care of this.

**Review by mafredri — APPROVED** [review]
This, and our scripts in general, could use a bit of an overhaul to be consistent with bash and non-bash features, but other than that, LGTM.

File: vault-jwt/run.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  install
+); then
+  echo "Failed to install Vault CLI."
+  exit 1
+fi
+rm -rf "$TMP"
+
+# Authenticate with Vault
+printf "🔑 Authenticating with Vault ...\n\n"
+echo "$${CODER_OIDC_ACCESS_TOKEN}" | vault write auth/$${VAULT_JWT_AUTH_PATH}/login role=$${VAULT_JWT_ROLE} jwt=-
```
[the remark this answers is not in the document]
Comment (matifali) [reply vault-jwt/run.sh:110]: Can we set up a linter to standardize these rules? Will shellcheck work?

[the remark this answers is not in the document]
Comment (mafredri) [reply vault-jwt/run.sh:110]: I _want_ to make shellcheck work, but mixing TF and `$${}` is not good for its blood pressure (if it has any).

I can try to make a proposal next week for how we could handle it, have some ideas.

[the remark this answers is not in the document]
Comment (matifali) [reply vault-jwt/run.sh:110]: Sounds great. Thank you :)

**Pull request merged**